tests/routers/test_auth.py

```
@@ -0,0 +1,179 @@
+
+def test_check_auth_status_no_users(client: TestClient):
+    """Test the /check endpoint when no users exist."""
+    response = client.get("/api/v1/auth/check")
+    assert response.status_code == 200
+    data = response.json()
+    assert data["setup_required"] is True
+    assert data["user_count"] == 0
+
+def test_check_auth_status_with_users(client: TestClient, db_session: Session):
+    """Test the /check endpoint when users exist."""
+    db_session.add(User(username="testuser", password_hash="..."))
```

Ruff (0.14.14)

[error] Possible hardcoded password assigned to argument: "password_hash"

(S106)